.github/workflows/build.yml

```diff
@@ -0,0 +1,36 @@
+name: Deploy the website
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: write
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+       
+      - name: Install Nix package manager
+        uses: DeterminateSystems/nix-installer-action@main
+
+      - name: Build the website
+        run: |
+          nix develop --command cargo run --verbose
+          nix develop --command cargo build --verbose
```

Comment:
run implies build, I think line is redundant?

Reply:
The `cargo build` doesn't actually build the site but the `cargo run` does. I've tested on my end. I recommend checking once by removing all the files inside `docs` directory to reproduce the issue. I thought it's a known issue.